Azure prereqs should include SKU restrictions (#19)
redpanda-data/cloud-docs: modules/get-started/pages/cluster-types/byoc/create-byoc-cluster-azure.adoc

Reviewer: I would add `for eg: for Microsoft.Compute run`

> == Prerequisites

Reviewer 2: Maybe I'm too old school...but this topic starts with 3 headings, and the first two have no content/text in them. Seems like maybe there should be a brief intro?

Author: added an intro

> === Check rpk

Reviewer 2: ```suggestion
=== Verify rpk version
```

> Confirm you have a minimum version of Redpanda `rpk` v24.1. See xref:manage:rpk/rpk-install.adoc[].

Reviewer 2: Interesting...the link to the rpk ref doc tells you how to install, etc, but nowhere in the doc page does it indicate what version of rpk doc you are looking at. It just gives you a command to install the "latest version of rpk". The version of rpk being installed on the page you referenced is 24.2--so the the linked rpk installation instructions seem like they are for 24.2, not 24.1. Perhaps link to the 24.1 version of this page instead? Also, have you considered automating this so that the link/rpk version requirement is automatically updated whenever Azure prereqs change to require a different version of rpk? Not sure if that is programmatically possible given it is likely a decision that simply comes down from PM or eng, but worth exploring to ensure this link/prereq is always up-to-date.

Author: @Feediver1: It doesn't need to be 24.1: that's just the minimum version. Instead of a link to Install rpk, I changed to See [rpk version](http://localhost:5002/redpanda-cloud/reference/rpk/rpk-version/) or [Introduction to rpk](http://localhost:5002/redpanda-cloud/manage/rpk/intro-to-rpk/).

=== Check your Azure subscription

> In the https://login.microsoftonline.com/[Azure Portal^], confirm that the dedicated subscription you will use with Redpanda has with the following: 

Reviewer 2: ```suggestion
In the https://login.microsoftonline.com/[Azure Portal^], confirm that the dedicated subscription you intend to use with Redpanda includes the following: 
```

Reviewer 2: ^^ This one is a "must" change

* Role: The Azure user must have the _Owner_ role in the subscription.

* Resources: The subscription must be registered for the following resource providers. See the https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types[Microsoft documentation^]. 

** Microsoft.Compute
** Microsoft.ManagedIdentity
** Microsoft.Storage
** Microsoft.KeyVault
** Microsoft.Network
+

> TIP: To check if an Azure resource provider is registered, run the following command using the Azure CLI or in the Azure Cloud Shell. For example, to check for Microsoft.Compute:

Reviewer 2: This is documenting how to use the Azure product...meaning that if the command/functionality ever changes, that you need to update this. We never document how to use other company's products.  Reminds me I need to update/add this to our Style Guide, as it isn't in there and it should be.

Author: @Feediver1 Are you saying we cannot document this?

Reviewer 2: Well, it goes against a long-standing convention. Once you break it, you open the door for other "breakage". Let's discuss offline.

+
``` 
az provider show -n Microsoft.Compute
```

* Feature: The subscription must be registered for Microsoft.Compute/EncryptionAtHost. See the https://learn.microsoft.com/en-us/azure/virtual-machines/linux/disks-enable-host-based-encryption-cli#prerequisites[Microsoft documentation^].

* Quota: The subscription must have the following quota in the region where you will use Redpanda. See the https://learn.microsoft.com/en-us/azure/quotas/view-quotas[Microsoft documentation^].

** Standard Lasv3-series vCPUs: 24
** Standard Dadsv5-series vCPUs: 8

=== Check Azure SKU restrictions

> Check if the Azure subscription has SKU restrictions for the VM sizes in the region where you will use Redpanda. Using the Azure CLI or in the Azure Cloud Shell, run:

Reviewer 2: ```suggestion
Check to ensure that the Azure subscription does not have any SKU restrictions for the VM sizes in the region where you intend to use Redpanda. Using the Azure CLI or in the Azure Cloud Shell, run:
```

----
az vm list-skus -l eastus2 --zone --size Standard_L8s_v3
az vm list-skus -l eastus2 --zone --size Standard_D2s_v3
az vm list-skus -l eastus2 --zone --size Standard_D2ads_v5
----

If you see restrictions, https://learn.microsoft.com/en-us/troubleshoot/azure/general/region-access-request-process[open a Microsoft support request^] to remove the restrictions. For example, checking the Standard_L8s_v3 VM SKU restriction in the eastus2 region for all zones could show:

[%nowrap,bash]
----
ResourceType     Locations    Name              Zones    Restrictions
---------------  -----------  ---------------   -------  ------------
virtualMachines  eastus2      Standard_L8s_v3   1,2,3    NotAvailableForSubscription, type: Zone, locations: eastus2, zones: 2,3
----

> == Create a BYOC cluster

Reviewer 2: Thoughts on starting with "After meeting the Prerequisites..." and link to the preceding section?
Many people will navigate to "Create a BYOC cluster" from a google search or link, and if they do and start here, they might miss the Prereq section.

[NOTE]
==== 
* If you plan to create a private network in your own virtual network, select the region where your Azure Virtual Network is located.

> * Currently, only multi-AZ (availability zones) is supported. Three AZs provide two backups in case one availability zone goes down.

Reviewer: Change this to convey " Multi-AZ is the default configuration. Single-AZ is supported as well"
